### ISSUE TITLE
`UnusedImports` false negative with two imports of the same name from different packages

### ISSUE THREAD

**Issue by DetachHead** [detekt/detekt#8120]
## Expected Behavior

`UnusedImports` error on `import kotlin.io.path.Path`. intellij's unused import inspection is able to detect it:

![Image](https://github.com/user-attachments/assets/9e39e596-944f-4d75-b98b-7717bb5a7bd1)

## Observed Behavior
no error

## Steps to Reproduce

```kt
import java.nio.file.Path // this import is used
import kotlin.io.path.Path // this one is not

typealias Foo = Path
```
## Your Environment

* Version of detekt used: 1.23.8
* Operating System and version: windows 10

**Comment by markoeltiger** [comment]
hello @3flex  can i get assigned for this?

**Comment by 3flex** [comment]
Sure but I would recommend waiting for the analysis API to become available and the rule rewritten first.

**Comment by DetachHead** [comment]
not stale

**Comment by BraisGabin** [comment]
@markoeltiger the Analysis API is a reality note so you can go for it if you still want

**Comment by markoeltiger** [comment]
@BraisGabin will start working on it !
